tests/read_mac/test_read_mac_metadata.py

```diff
@@ -0,0 +1,146 @@
+import pytest
+import logging
+
+from tests.common.utilities import wait
+from tests.common.utilities import wait_until
+from tests.common.plugins.sanity_check import checks
+from tests.common.plugins.loganalyzer.loganalyzer import LogAnalyzer
+from tests.common.helpers.assertions import pytest_assert
+from datetime import datetime
+
+
+logger = logging.getLogger(__name__)
+
+BINARY_FILE_ON_LOCALHOST = "/tmp/update_image.bin"
+BINARY_FILE_ON_DUTHOST = "/tmp/mellanox.bin"
+
+pytestmark = [
+    pytest.mark.topology('any')
+]
+
+@pytest.fixture(scope='function')
+def cleanup_read_mac(duthost, localhost):
+    yield
+    logger.info('Remove temporary images')
+    duthost.shell("rm -rf {}".format(BINARY_FILE_ON_DUTHOST))
+    localhost.shell("rm -rf {}".format(BINARY_FILE_ON_LOCALHOST))
+
+    backup_minigraph_exist = duthost.stat(path = "/etc/sonic/minigraph.xml.backup")["stat"]["exists"]
+    if backup_minigraph_exist:
+        logger.info("Apply minigraph from backup")
+        duthost.shell("mv /etc/sonic/minigraph.xml.backup /etc/sonic/minigraph.xml")
+        duthost.shell("config load_minigraph -y")
+        duthost.shell("config save -y")
+
+class ReadMACMetadata():
+    def __init__(self, request):
+        self.iteration = request.config.getoption("--iteration")
+        self.image1 = request.config.getoption("--image1")
+        self.image2 = request.config.getoption("--image2")
+        self.minigraph1 = request.config.getoption("--minigraph1")
+        self.minigraph2 = request.config.getoption("--minigraph2")
+
+        if not self.image1:
+            pytest.fail("Please specify --image1")
+        if not self.image2:
+            pytest.fail("Please specify --image2")
```

Comment:
Can checking availability of option `image1` and `image2` be moved to `tests/read_mac/conftest.py`?  While calling `parser.addoption`, add argument `required=True` should be able to specify that this option is required.

Reply:
Done